### PR TITLE
TASK: Add tests for FusionService::generateFusionForNodeType

### PR DESCRIPTION
**Pull request by nezaniel**

### CODE REVIEW

**Comment by mficzel** [comment]
This is a follow op on https://github.com/neos/neos-development-collection/issues/767 to add some functional tests.

File: TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+/**
+ * Tests for the TypoScriptService
+ */
+class TypoScriptServiceTest extends FunctionalTestCase
+{
+    /**
+     * @var string the Nodes fixture
+     */
+    protected $fixtureFileName = 'Fixtures/NodeTypes.yaml';
```
Comment (gerhard-boden) [inline TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php]: use a constant here

File: TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        $this->typoScriptService = $this->objectManager->get(TypoScriptService::class);
+
+        $this->expectedPrototypeGenerator = $this->objectManager->get(TestablePrototypeGenerator::class);
+
+        $this->yamlParser = $this->objectManager->get(YamlParser::class);
+
+        $this->originalNodeTypeManager = $this->objectManager->get(NodeTypeManager::class);
+
+        $this->mockNodeTypeManager = clone ($this->originalNodeTypeManager);
+        $this->mockNodeTypeManager->overrideNodeTypes($this->yamlParser->parse(file_get_contents(__DIR__ . '/Fixtures/NodeTypes.yaml')));
```
Comment (gerhard-boden) [inline TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php]: read the constant from above here

File: TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        $this->invokeGenerateTypoScriptForNodeType('TYPO3.Neos:NodeTypeWithInvalidFusionPrototypeGenerator');
+    }
+
+    /**
+     * @test
+     */
+    public function generateTypoScriptForNodeDoesNotUseFusionPrototypeGeneratorWithoutConfiguration()
+    {
+        $this->invokeGenerateTypoScriptForNodeType('TYPO3.Neos:NodeTypeWithoutFusionPrototypeGenerator');
+        $this->assertSame(0, $this->expectedPrototypeGenerator->getCallCount());
```
Comment (gerhard-boden) [inline TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php]: can we use php unit instruments to measure how often something was called?

File: TYPO3.Neos/Tests/Functional/Domain/Service/Fixtures/NodeTypes.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+'TYPO3.Neos:NodeTypeWithInvalidFusionPrototypeGenerator':
+  options:
+    fusion:
+      prototypeGenerator: TYPO3\Neos\Domain\Service\InexsitentPrototypeGenerator
```
Comment (johannessteu) [inline TYPO3.Neos/Tests/Functional/Domain/Service/Fixtures/NodeTypes.yaml]: InexistentPrototypeGenerator (typo)

File: TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        parent::setUp();
+
+        $this->typoScriptService = $this->objectManager->get(TypoScriptService::class);
+
+        $this->expectedPrototypeGenerator = $this->objectManager->get(TestablePrototypeGenerator::class);
+
+        $this->yamlParser = $this->objectManager->get(YamlParser::class);
+
+        $this->originalNodeTypeManager = $this->objectManager->get(NodeTypeManager::class);
```
Comment (johannessteu) [inline TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php]: remove the whitespace lines :)

File: TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php (hunk 3 above)
Comment (johannessteu) [inline TYPO3.Neos/Tests/Functional/Domain/Service/TypoScriptServiceTest.php]: probably we can, but I think this implementation is fine as is, really easy and straight forward to understand. so fine for me :)

**Comment by kitsunet** [comment]
Test failure is unrelated, but should be fixed in a separate change.

**Pull request merged**